addons/hr_expense/models/hr_expense.py

```diff
@@ -822,6 +823,13 @@ def _compute_can_reset(self):
         for sheet in self:
             sheet.can_reset = is_expense_user if is_expense_user else sheet.employee_id.user_id == self.env.user
 
+    @api.depends_context('uid')
+    def _compute_can_approve(self):
+        is_approver = self.user_has_groups('hr_expense.group_hr_expense_team_approver, hr_expense.group_hr_expense_user')
+        is_manager = self.user_has_groups('hr_expense.group_hr_expense_manager')
+        for sheet in self:
+            sheet.can_approve = is_manager if is_manager else is_approver and sheet.employee_id.user_id != self.env.user
```

Comment:
Similar to 
```suggestion
            sheet.can_approve = is_manager or is_approver and sheet.employee_id.user_id != self.env.user
```
right ?